cmake/CheckAtomic.cmake

```diff
@@ -0,0 +1,41 @@
+# - Try to find if 64-bits atomics need -latomic linking
+# Once done this will define
+#  HAVE_CXX_ATOMICS_WITHOUT_LIB - Whether atomic types work without -latomic
+
+INCLUDE(CheckCXXSourceCompiles)
```

Comment:
Please follow existing coding style:
* lower-case: command names (`include()`, `check_cxx_source_compiles()`)
* upper-case: variable names (`VARNAME`), boolean values (`TRUE`)
* indentation: 4 spaces
* no arguments repeated in end commands (`endfunction()`)